src/Core/Application/Common/Specification/SpecificationBuilderExtensions.cs

```diff
@@ -79,29 +81,178 @@ public static ISpecificationBuilder<T> PaginateBy<T>(this ISpecificationBuilder<
     }
 
-    private static void AddSearchPropertyByKeyword<T>(this ISpecificationBuilder<T> specificationBuilder, Expression propertyExpr, ParameterExpression paramExpr, string keyword)
+    private static MemberExpression GetMemberExpression(string propertyName, ParameterExpression parameter)
+    {
+        Expression mapProperty = parameter;
+        foreach (string member in propertyName.Split('.'))
+        {
+            mapProperty = Expression.PropertyOrField(mapProperty, member);
+        }
+
+        return (MemberExpression)mapProperty;
+    }
+
+    private static BinaryExpression GetBinaryExpression(MemberExpression memberExpression, ConstantExpression constantExpression, ConstantExpression constantExpressionAux, Filter filter)
+    {
+        return filter.Operator switch
+        {
+            FilterOperator.EQ => Expression.Equal(memberExpression, constantExpression),
+            FilterOperator.NEQ => Expression.NotEqual(memberExpression, constantExpression),
+            FilterOperator.LT => Expression.LessThan(memberExpression, constantExpression),
+            FilterOperator.LTE => Expression.LessThanOrEqual(memberExpression, constantExpression),
+            FilterOperator.GT => Expression.GreaterThan(memberExpression, constantExpression),
+            FilterOperator.GTE => Expression.GreaterThanOrEqual(memberExpression, constantExpression),
+            FilterOperator.BETWEEN => Expression.AndAlso(Expression.GreaterThanOrEqual(memberExpression, constantExpression), Expression.LessThanOrEqual(memberExpression, constantExpressionAux)),
+            _ => throw new ArgumentException("operatorSearch is not valid.", nameof(filter.Operator)),
+        };
+    }
+
+    private static string GetStringFromJsonElement(object value) => ((JsonElement)value).GetString()!;
+
+    private static BinaryExpression GetBinaryExpressionFromFilter(Filter filter, ParameterExpression parameter)
+    {
+        MemberExpression mapProperty = GetMemberExpression(filter.Field, parameter);
+
+        (ConstantExpression value, ConstantExpression valueAx) = GetConstantExpressionFromFilter(mapProperty, filter);
+
+        var bExpresion = GetBinaryExpression(mapProperty, value, valueAx, filter);
+        if (filter.Filters.Any())
+        {
+            bExpresion = AddFilters(filter.Filters, parameter, bExpresion);
+        }
+
+        return bExpresion;
+    }
+
+    private static (ConstantExpression cExpresion, ConstantExpression cExpresionAux) GetConstantExpressionFromFilter(MemberExpression mapProperty, Filter filter)
+    {
+        ConstantExpression cExpresionAux = default!;
+
+        ConstantExpression cExpresion;
+        if (mapProperty.Type.IsEnum)
+        {
+            string? stringEnum = GetStringFromJsonElement(filter.Value!);
+
+            if (!Enum.TryParse(mapProperty.Type, stringEnum, true, out object? valueparsed)) throw new CustomException("Value {0} is not valid for {1}");

```

Comment:
CustomException?
And where are the values for the holes in the fromatstring?

same thing few lines lower...